File: examples/python/kafka-producer-consumer/producer.py
import json
import os
import time
from kafka import KafkaProducer

brokers = os.getenv("KAFKA_BROKERS", "localhost:9092").split(",")
producer = KafkaProducer(bootstrap_servers=brokers, value_serializer=lambda v: json.dumps(v).encode("utf-8"))
producer.send("app-events", {"id": int(time.time()), "event": "python.produced"})
producer.flush()
print("[producer] sent")

File: examples/nodejs/kafka-producer-consumer/producer.js
import { Kafka } from 'kafkajs';

const brokers = (process.env.KAFKA_BROKERS || 'localhost:9092').split(',');
const kafka = new Kafka({ clientId: 'node-producer', brokers });
const producer = kafka.producer();

await producer.connect();
await producer.send({ topic: 'app-events', messages: [{ value: JSON.stringify({ id: Date.now(), event: 'node.produced' }) }] });
console.log('[producer] kafka message sent');
await producer.disconnect();
